crates/astria-sequencer-relayer/src/finalization_pipeline/pipeline_item.rs

```diff
@@ -0,0 +1,87 @@
+use astria_sequencer_types::SequencerBlockData;
+use tendermint::Hash;
+
+use super::{
+    Head,
+    HeadCandidate,
+};
+
+#[derive(Clone, Copy, Default, Debug)]
+enum State {
+    #[default]
+    Fork,
+    Canonical,
+}
+
+#[derive(Default, Debug)]
+pub(crate) struct PipelineItem {
+    block: HeadCandidate,
+    state: State,
+}
+
+impl From<HeadCandidate> for PipelineItem {
+    fn from(block: HeadCandidate) -> Self {
+        Self {
+            block,
+            state: State::default(),
+        }
+    }
+}
+
+impl TryInto<SequencerBlockData> for PipelineItem {
+    type Error = ();
+
+    fn try_into(self) -> Result<SequencerBlockData, Self::Error> {
+        self.block.try_into()
+    }
+}
+
+impl PipelineItem {
```

Comment:
I'm unclear on the purpose of `PipelineItem`, since blocks are already sorted within the `FinalizationPipeline` as finalized or not, this seems unnecessary

Reply:
it is used for destructing blocks in the finalisation pipeline into `Head` and `SequencerBlockData`